src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java

```diff
@@ -49,7 +49,7 @@
  * All AbstractBlockHut[something] should extend this class.
  */
 @SuppressWarnings("PMD.ExcessiveImports")
-public abstract class AbstractBlockHut<B extends AbstractBlockHut<B>> extends AbstractBlockMinecolonies<B> implements IBuilderUndestroyable, IAnchorBlock
+public abstract class AbstractBlockHut<B extends AbstractBlockHut<B>> extends AbstractBlockMinecolonies<B> implements IBuilderUndestroyable, IAnchorBlock, IBlockHut
```

Comment:
Can't we move IHutBlock to MineColonies, and in MineColonies extend structurename and use our own version which does the hut check or so?

Reply:
If we go as far as Improving this, then we should remove structurize having to know about huts

Reply:
If I were to write this on MC's side, I would just check for membership in the AbstractHutBlock class, and not worry about the interface at all.  However, this check only matters for schematics, which seems to be why we would want it in Structurize.

Reply:
Well, it's only for schematics, but who defines special treatment of schematics should be the mInecolonies side, and not the structurize side

Reply:
Alright. I'll see what I can on that front

Reply:
Hold on. I'm gonna cancel this PR and we just need to convince Orion to wait on these changes. Section/style/level should all be in the Metadata changes. This is adding a 'fix' for something that really will be fixed at a later date in a better way.

Reply:
It would be nice if the levelled decorations could have a level selector similar to huts (would save a lot of "why is it building level 1 instead of level 3" questions).  Having said that, that's tricky with the current UI arrangement and you probably wouldn't want to treated the levelled decos as "huts" as far as appearing individually in the first selector, so it's not as easy as that sounds.